src/worker/hwmonitor.rs

```diff
@@ -1,21 +1,37 @@
 use serde::{Deserialize, Serialize};
-use systemstat::{saturating_sub_bytes, Platform, System};
+use std::thread;
+use std::time::Duration;
+use systemstat::{BTreeMap, CPULoad, Memory, Network, Platform, System};
 
-#[derive(Deserialize, Serialize, Debug)]
+#[derive(Deserialize, Serialize, Debug, Default)]
 pub struct WorkerHwState {
-    pub worker_cpu_usage: i8,
-    pub worker_memory_usage: i8,
-    pub worker_network_usage: i8,
+    pub worker_cpu_usage: Option<CPULoad>,
+    pub worker_memory_usage: Option<Memory>,
+    pub worker_network_usage: Option<BTreeMap<String, Network>>,
 }
 
-//fixme: what should I do with read error? read error shouldn't make it to the average
-// should a read error lead to a none in the hwState optional and a warning?
-pub fn get_worker_hw_state() -> Option<WorkerHwState> {
+pub fn get_worker_hw_state() -> WorkerHwState {
     let sys = System::new();
+    let mut hw_state: WorkerHwState = Default::default();
+    match sys.networks() {
+        Ok(networks) => hw_state.worker_network_usage = Some(networks),
+        Err(x) => {
+            log::warn!("\nNetwork usage read error: {}", x)
+        }
+    }
 
-    Option::from(WorkerHwState {
-        worker_cpu_usage: 0,
-        worker_memory_usage: 0,
-        worker_network_usage: 0,
-    })
+    match sys.memory() {
+        Ok(mem) => hw_state.worker_memory_usage = Some(mem),
+        Err(x) => log::warn!("\nMemory usage read error: {}", x),
+    }
+
+    match sys.cpu_load_aggregate() {
+        Ok(cpu) => {
+            thread::sleep(Duration::from_secs(1));
```

Comment:
I have not figured out the implications of this sleep

Reply:
The sleep definitely cannot be here.

It would be ideal if we could just get the immediate use of all available cores. It seems that `systemstat` doesn't provide it though. It could be solved by converting this function to `async` and use `tokio::wait_for(Duration::from_secs(1))` before reading the CPU usage.